src/sparsezoo/utils/download.py

```diff
@@ -12,209 +12,385 @@
+        self._lock = threading.Lock()
+
+    def is_range_header_supported(self) -> bool:
+        """Check if chuck download is supported"""
+        response = requests.head(
+            self.url, headers={"Accept-Encoding": "identity"}, allow_redirects=True
+        )
+        if response.headers.get("accept-ranges") is not None:
+            return True
+        return False
+
+    def get_file_size(self):
+        """Get file size of the requested file"""
+        response = requests.head(
+            self.url, headers={"Accept-Encoding": "identity"}, allow_redirects=True
+        )
+        file_size = int(response.headers.get("Content-Length", -1))
+        if file_size > 0:
+            return file_size
+        raise ValueError(f"Invalid download URL: {self.url}")
+
+    def queue_chunk_download_jobs(self) -> None:
+        """
+        Queues two jobs: file chunk download and combining the chunks.
+
+        This method calculates the number of download jobs needed based on the size of
+         the file to be downloadedand the size of each chunk. It creates download jobs
+         for each chunk, where each job is responsible for downloading a specific byte
+         range of the file. After queuing all the chunk download jobs, it queues an
+         additional job to combine these chunks into the final file.
+
+        Each download job is added to a JobQueue with a descrition (shown in the
+         progress bar), and the jobs areconfigured with the necessary parameters,
+         including the path where the chunk will be saved andthe 'Range' header to
+         download the specified byte range. The method ensures that the byte ranges
+         are calculated correctly to cover the entire file, even if the file size is
+         not a multiple of the chunk size.
+
+        After all chunk download jobs are queued, if at least one chunk download job
+         exists, a final job is queued to combine all downloaded chunks into the final
+         file and deletethe individual chunk files. This combining job is added to a
+         separate JobQueue with its own descriptive label.
+
+        Side effects:
+            - Queues jobs in the instance's `job_queues` attribute, which should be a
+               Queue of JobQueues.
+            - Utilizes the `download_file` method to download each chunk.
+            - Utilizes the `get_chunk_file_path` method to determine the path for each
+               chunk's temporary file.
+            - Utilizes the `combine_chunks_and_delete` method as the final step to
+               combine all chunks and clean up.
+
+        Note:
+            This method does not start the download jobs; it only queues them.
+            The jobs need to be executed by a worker or scheduler that processes the
+             queued JobQueues.
+        """
+        download_jobs: Queue[Job] = JobQueue(description="Downloading Chunks")
+        num_download_jobs = self.file_size // self.chunk_bytes + int(
```

Comment:
what's the logic here?

Reply:
It computes the number of download jobs wrt the chunk size

Reply:
I used ceil, logic is simpler now